src/rx-query.ts

```diff
@@ -490,6 +486,121 @@ export class RxQueryBase<
         this._limitBufferSize = bufferSize;
         return this;
     }
+
+    enablePersistentQueryCache(backend: QueryCacheBackend) {
+        this._persistentQueryCacheBackend = backend;
+        this._persistentQueryCacheLoaded = this._restoreQueryCacheFromPersistedState();
+        return this;
+    }
+
+    private async _restoreQueryCacheFromPersistedState() {
+        if (!this._persistentQueryCacheBackend) {
+            // no cache backend provided, do nothing
+            return;
+        }
+
+        if (this._persistentQueryCacheResult) {
+            // we already restored the cache once, no need to run twice
+            return;
+        }
+
+        if (this.mangoQuery.skip) {
+            console.error('The persistent query cache only works on non-skip queries.');
+            return;
+        }
+
+        const persistentQueryId = this.persistentQueryId();
+        const value = await this._persistentQueryCacheBackend.getItem<string[] | string>(`qc:${persistentQueryId}`);
+        if (!value) {
+          return;
+        }
+
+        const lwt = (await this._persistentQueryCacheBackend.getItem(`qc:${persistentQueryId}:lwt`)) as string ?? RX_META_LWT_MINIMUM;
+
+        const primaryPath = this.collection.schema.primaryPath;
+
+        this._persistentQueryCacheResult = value ?? undefined;
+        this._persistentQueryCacheResultLwt = lwt;
+
+        // if this is a regular query, also load documents into cache
+        if (Array.isArray(value) && value.length > 0) {
+            const persistedQueryCacheIds = new Set(this._persistentQueryCacheResult);
+
+            let docsData: RxDocumentData<RxDocType>[] = [];
+
+            // query all docs updated > last persisted, limit to an arbitrary 1_000_000 (10x of what we consider our largest library)
+            const {documents: changedDocs} = await this.collection.storageInstance.getChangedDocumentsSince(
+              1_000_000,
+              // make sure we remove the monotonic clock (xxx.01, xxx.02) from the lwt timestamp to avoid issues with
+              // lookups in indices (dexie)
+              {id: '', lwt: Math.floor(Number(lwt))}
+            );
+
+            for (const changedDoc of changedDocs) {
```

Comment:
i wonder if there's some number of changedDocs that we just want to give up and re-exec the query. i'm imagining a user bulk updating thousands of docs....

Reply:
You would still need to explicitly enable the cache for such queries, but it is not too hard to change if we want it.